### PR TITLE
#135 Redirect application to first incomplete page

### PR DESCRIPTION
**Pull request by CarlosNZ**
Implements #135 -- just a fairly small extension to #194 (so you'll need [that PR's back-end branch](https://github.com/openmsupply/application-manager-server/pull/150) to test against, if it's not already merged)

This PR does two additional things:
1. Ensures that any empty response (null, undefined or empty string) all get set to "invalid" (for required questions) when re-validating all (change to `revalidateAll.tsx`). This ensures that when attempting to load Summary, you'll be taken back to first incomplete page (strict check), not just page with an invalid element (loose check).
2. Loading "Draft" application re-directs user to first invalid/incomplete page (instead of section 1, page 1) -- uses the same `revalidateAll` and `firstErrorLocation` functions as #194. (This does NOT write validity back to database, so `null` validations will persist in each element.

### CODE REVIEW

File: src/containers/Application/ApplicationPageWrapper.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ const ApplicationPageWrapper: React.FC = () => {
   // 1 - ProcessRedirect: Will redirect to summary in case application is SUBMITTED
   // 2 - Set the current section state of the application
   useEffect(() => {
-    if (isApplicationLoaded) {
+    if (elementsState && responsesByCode) {
       processRedirect({
```
Comment (nmadruga) [inline src/containers/Application/ApplicationPageWrapper.tsx:83]: Don't need to use await here?

File: src/containers/Application/ApplicationPageWrapper.tsx [hunk 2]
```diff
@@ -290,14 +292,26 @@ function processRedirect(appState: any): void {
...
+
+    if (revalidate.validityFailures.length > 0) {
+      const { firstErrorSectionCode, firstErrorPage } = getFirstErrorLocation(
+        revalidate.validityFailures,
+        elementsState as ApplicationElementStates
+      )
+      push(`/application/${serialNumber}/${firstErrorSectionCode}/Page${firstErrorPage}`)
+    } else {
+      const firstSection = templateSections[0].code
+      replace(`/application/${serialNumber}/${firstSection}/page1`)
```
Comment (nmadruga) [inline src/containers/Application/ApplicationPageWrapper.tsx:313]:
```suggestion
      replace(`/application/${serialNumber}/${firstSection}/Page1`)
```

**Review by nmadruga — COMMENTED** [review]
I'll be testing this one now. Getting all pieces together looks good.

Make just 2 suggestions on this one for you to have a look please

File: src/containers/Application/ApplicationPageWrapper.tsx (hunk 1 above)
Comment (CarlosNZ) [reply src/containers/Application/ApplicationPageWrapper.tsx:83]: I don't think so, cos if there is a redirect required, then the called method does the `push`, so it'll over-rule anything else in progress. And if there's no redirect, then we don't need to wait for anything.

Comment (nmadruga) [reply src/containers/Application/ApplicationPageWrapper.tsx:83]: Ah! Got it. That's good.

**Review by nmadruga — APPROVED** [review]
LGTM

**Pull request merged**